x-pack/plugins/security_solution/server/lib/telemetry/tasks/prebuilt_rule_alerts.ts

```diff
@@ -47,53 +52,56 @@ export function createTelemetryPrebuiltRuleAlertsTaskConfig(maxTelemetryBatch: n
             : ({} as ESLicense | undefined);
         const packageInfo =
           packageVersion.status === 'fulfilled' ? packageVersion.value : undefined;
+        const index = alertsIndex.status === 'fulfilled' ? alertsIndex.value : undefined;
 
-        const { events: telemetryEvents, count: totalPrebuiltAlertCount } =
-          await receiver.fetchPrebuiltRuleAlerts();
-
-        sender.getTelemetryUsageCluster()?.incrementCounter({
-          counterName: 'telemetry_prebuilt_rule_alerts',
-          counterType: 'prebuilt_alert_count',
-          incrementBy: totalPrebuiltAlertCount,
-        });
-
-        if (telemetryEvents.length === 0) {
-          tlog(logger, 'no prebuilt rule alerts retrieved');
-          await sender.sendOnDemand(TASK_METRICS_CHANNEL, [
-            createTaskMetric(taskName, true, startTime),
-          ]);
+        if (index === undefined) {
+          tlog(logger, `alerts index is not ready yet, skipping telemetry task`);
           return 0;
         }
 
-        const processedAlerts = telemetryEvents.map(
-          (event: TelemetryEvent): TelemetryEvent =>
-            copyAllowlistedFields(filterList.prebuiltRulesAlerts, event)
-        );
+        let fetchMore = true;
+        let searchAfterValue: SortResults | undefined;
+        let pitId = await receiver.openPointInTime(index);
+
+        while (fetchMore) {
+          const { moreToFetch, newPitId, searchAfter, alerts } =
```

Comment:
Should we have a early exit and log here if alerts is empty? I don't believe anything following in this block will fail since they are just loops over it, but perhaps just as a safe guard for future changes?

Reply:
The `moreToFetch` returned value is determined by `fetchMore = numOfHits > 0 && numOfHits < 1_000;` in `fetchPrebuiltRuleAlertsBatch`  so when it returns on its last iteration it should always have results and be `false`.  But I understand that it could be a future footgun so I will introduce a log and break out of the loop.

Reply:
Thanks. Updated in https://github.com/elastic/kibana/pull/170334/commits/2562b6f5e7536779d9dde6a8bde55e6886cbff56